specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-11-15/restorable.json

```diff
@@ -0,0 +1,1815 @@
+          },
+          {
+            "$ref": "#/parameters/locationParameter"
+          },
+          {
+            "$ref": "#/parameters/instanceIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/RestorableGremlinDatabasesListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "cosmos-db.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/locations/{location}/restorableDatabaseAccounts/{instanceId}/restorableGraphs": {
+      "get": {
+        "operationId": "RestorableGremlinGraphs_List",
+        "x-ms-examples": {
+          "CosmosDBRestorableMongodbCollectionList": {
+            "$ref": "./examples/CosmosDBRestorableGremlinGraphList.json"
+          }
+        },
+        "description": "Show the event feed of all mutations done on all the Azure Cosmos DB Gremlin graphs under a specific database. This helps in scenario where container was accidentally deleted. This API requires 'Microsoft.DocumentDB/locations/restorableDatabaseAccounts/.../read' permission",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/locationParameter"
+          },
+          {
+            "$ref": "#/parameters/instanceIdParameter"
+          },
+          {
+            "$ref": "#/parameters/restorableGremlinDatabaseRidParameter"
+          },
+          {
+            "$ref": "#/parameters/restoreStartTime"
+          },
+          {
+            "$ref": "#/parameters/restoreEndTime"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/RestorableGremlinGraphsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "cosmos-db.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/locations/{location}/restorableDatabaseAccounts/{instanceId}/restorableGremlinResources": {
```

Comment:
Please consider modeling it as POST (e.g. listRestorableGremlinResources). Otherwise, GET needs to return the required properties for ARM resources like type, id, name.

Reply:
Hi Jin,
We have similar API already existing for Sql/MongoDB (https://github.com/Azure/azure-rest-api-specs/blob/569bb5b5926008180266346d4af44647363446d4/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-11-15/restorable.json#L305).
We are just adding support for Tables and Gremlin for same restorableResources Api.

Reply:
Okay. I will approve since there are already existing APIS using the patten and it is not a blocker.